src/Elliptic/Systems/Poisson/Equations.cpp

```diff
@@ -23,59 +23,31 @@
 namespace Poisson {
 
 template <size_t Dim>
-void ComputeFirstOrderOperatorAction<Dim>::apply(
-    const gsl::not_null<Scalar<DataVector>*> operator_for_field_source,
+void euclidean_fluxes(
     const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
-        operator_for_auxiliary_field_source,
-    const tnsr::i<DataVector, Dim, Frame::Inertial>& grad_field,
-    const tnsr::I<DataVector, Dim, Frame::Inertial>& auxiliary_field,
-    const Mesh<Dim>& mesh,
-    const InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
-        inverse_jacobian) noexcept {
-  auto div_vars =
-      make_with_value<Variables<tmpl::list<Tags::AuxiliaryField<Dim>>>>(
-          auxiliary_field, 0.);
-  get<Tags::AuxiliaryField<Dim>>(div_vars) = auxiliary_field;
-  // Tensors don't support math operations yet, so we have to `get` the
-  // DataVector for the sign flip
-  get(*operator_for_field_source) =
-      -1. * get(get<::Tags::div<Tags::AuxiliaryField<Dim>>>(
-                divergence(div_vars, mesh, inverse_jacobian)));
+        flux_for_field,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& field_gradient) noexcept {
   for (size_t d = 0; d < Dim; d++) {
-    operator_for_auxiliary_field_source->get(d) =
-        grad_field.get(d) - auxiliary_field.get(d);
+    flux_for_field->get(d) = field_gradient.get(d);
   }
 }
 
 template <size_t Dim>
-void ComputeFirstOrderNormalDotFluxes<Dim>::apply(
-    const gsl::not_null<Scalar<DataVector>*> normal_dot_flux_for_field,
-    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
-        normal_dot_flux_for_auxiliary_field,
-    const Scalar<DataVector>& field,
-    const tnsr::I<DataVector, Dim, Frame::Inertial>& auxiliary_field,
-    const tnsr::i<DataVector, Dim, Frame::Inertial>&
-        interface_unit_normal) noexcept {
-  // The minus sign in the equation is canceled by the one in `lift_flux`
-  get(*normal_dot_flux_for_field) =
-      get<0>(interface_unit_normal) * get<0>(auxiliary_field);
-  for (size_t d = 1; d < Dim; d++) {
-    get(*normal_dot_flux_for_field) +=
-        interface_unit_normal.get(d) * auxiliary_field.get(d);
-  }
-
-  // The minus sign is to cancel the one in `lift_flux`
+void auxiliary_fluxes(gsl::not_null<tnsr::IJ<DataVector, Dim, Frame::Inertial>*>
+                          flux_for_gradient,
+                      const Scalar<DataVector>& field) noexcept {
```

Comment:
How would you handle an elliptic equation whose fields include a vector field? I'm just a little confused why this is called `field` instead of `scalar_field`

Reply:
This is the implementation for the Poisson system, so I'm working with the (scalar) `Poisson::Tags::Field` and its gradient. An example for an elliptic system that uses a vector field is the Elasticity system that I'll add once these preparatory PRs are in. It implements its own fluxes and sources for the `Elasticity::Tags::Displacement` vector field and the `Elasticity::Tags::Strain` symmetric rank-2 tensor.